core/src/main/java/io/grpc/internal/PickFirstLeafLoadBalancer.java

```diff
@@ -324,19 +439,18 @@ private Subchannel createNewSubchannel(SocketAddress addr) {
             new EquivalentAddressGroup(addr)))
         .addOption(HEALTH_CONSUMER_LISTENER_ARG_KEY, hcListener)
             .build());
+    if (subchannel == null) {
```

Comment:
If the helper is created correctly.  I would prefer not to trust something that we don't completely control and do the check.  However, I am now throwing an exception rather than returning null.